apps/dashboard/app/models/project.rb

```diff
@@ -72,6 +74,7 @@ def templates
   validates :icon, format: { with: %r{\Afa[bsrl]://[\w-]+\z}, allow_blank: true, message: :format }, on: [:create, :update]
   validate :project_directory_invalid, on: [:create, :update]
   validate :project_directory_exist, on: [:create]
+  validate :project_template_invalid, on: [:create]
```

Comment:
This isn't necessarily specific to this PR - but why are we using `:create` here? There's no `script#create` API.

How does this work when nobody is calling or defined a `create` API?

Reply:
I think I see, we call `valid?(:create)` somewhere. We can defer this, but that seems confusing.

Reply:
I can update it to `:save` to be consistent with the model method call when validating

Reply:
Let's do it in another PR - I think I'm good with this.